src/config/passport.ts

```diff
@@ -75,7 +75,30 @@ export const isAuthenticated = (req: Request, res: Response, next: NextFunction)
  * Authorization Required middleware.
  */
 export const isAdmin = (req: Request, res: Response, next: NextFunction) => {
-  if ((req.user as User)?.admin) {
+  if ((req.user as User)?.role == RoleType.ADMIN) {
+    next()
+  } else {
+    if (req.method !== 'GET') {
+      return res.sendStatus(403)
+    }
+    res.render('error/forbidden')
+  }
+}
+
+export const isTicketAdmin = (req: Request, res: Response, next: NextFunction) => {
+  if ((req.user as User)?.role == RoleType.TICKET_ADMIN) {
+    next()
+  } else {
+    if (req.method !== 'GET') {
+      return res.sendStatus(403)
+    }
+    res.render('error/forbidden')
+  }
+}
+
+export const isTicketOrSiteAdmin = (req: Request, res: Response, next: NextFunction) => {
+  const role = (req.user as User)?.role
+  if (role == RoleType.ADMIN || role == RoleType.TICKET_ADMIN) {
```

Comment:
What do you think about a universal `requireRoles(...roles: RoleType[])` function that takes the required roles as an argument and calls `next()` if the user has any one of them and renders forbidden otherwise?